src/Microsoft.SqlTools.ServiceLayer/SchemaCompare/SchemaCompareService.cs

```diff
@@ -231,6 +232,53 @@ await requestContext.SendResult(new ResultStatus()
             }
         }
 
+        /// <summary>
+        /// Handles request for schema compare publish database changes script
+        /// </summary>
+        /// <returns></returns>
+        public async Task HandleSchemaComparePublishProjectChangesRequest(SchemaComparePublishProjectChangesParams parameters, RequestContext<SchemaComparePublishProjectResult> requestContext)
+        {
+            SchemaComparePublishProjectChangesOperation operation = null;
+            try
+            {
+                SchemaComparisonResult compareResult = schemaCompareResults.Value[parameters.OperationId];
+                operation = new SchemaComparePublishProjectChangesOperation(parameters, compareResult);
+
+                SqlTask sqlTask = null;
+                TaskMetadata metadata = new TaskMetadata();
+                metadata.TaskOperation = operation;
+                metadata.ProjectPath = parameters.TargetProjectPath;
+                metadata.FolderStructure = parameters.TargetFolderStructure;
+                metadata.Name = SR.PublishChangesTaskName;
+
+                sqlTask = SqlTaskManagerInstance.CreateAndRun<SqlTask>(metadata);
+
+                // very ugly... needs fixing
+                while (operation.PublishResult == null) { };
+
+                await requestContext.SendResult(new SchemaComparePublishProjectResult()
+                {
+                    ChangedFiles = operation.PublishResult.ChangedFiles,
+                    AddedFiles = operation.PublishResult.AddedFiles,
+                    DeletedFiles = operation.PublishResult.DeletedFiles,
+                    Success = true,
+                    ErrorMessage = operation.ErrorMessage
+                });
+            }
+            catch (Exception e)
+            {
+                Logger.Write(TraceEventType.Error, "Failed to publish schema compare database changes. Error: " + e);
+                await requestContext.SendResult(new SchemaComparePublishProjectResult()
+                {
+                    ChangedFiles = Array.Empty<string>(),
```

Comment:
can these just be [] instead of Array.Empty<string>()?

Reply:
This gave me an error. I originally had `new string [] {}` but Visual Studio gave me the suggestions to use `Array.Empty`. Not sure which one is better.

Reply:
ah ok I was just getting confused with how this can be done in Typescript. This way should be fine :)